### PR TITLE
Fix for nested lists of models in schema

### PR DESCRIPTION
**Pull request by ilongin**
The problem was if we had `list[list[SomeModel]]` as a field in our custom model as flattening worked only for one level of lists with models.

This PR fixes `flatten(obj)` function to be able to flatten any number of nested lists and adds tests.

### CODE REVIEW

**Review by dtulga — COMMENTED** [review]
This looks good to me, but I see that the `test_custom_model_with_nested_lists` test is failing in ClickHouse with Pydantic ValidationErrors. That would ideally be fixed before merging.

File: src/datachain/lib/convert/flatten.py [hunk 1]
```diff
@@ -42,16 +42,20 @@ def flatten_list(obj_list):
 
 
 def _flatten_fields_values(fields, obj: BaseModel):
+    def _flatten_list(value):
```
Comment (dmpetrov) [inline src/datachain/lib/convert/flatten.py:45]: Please extract this func from func.

**Review by dmpetrov — APPROVED** [review]
Is this an alternative implementation to #192?

**Comment by ilongin** [comment]
> Is this an alternative implementation to #192? Please make sure it solves #182 and close 192.

Yes, I somehow missed your PR and saw that there was no connected PR to the issue itself and issue was unassigned so I took it.

**Pull request merged**